src/main/app/form/parser.ts

```diff
@@ -8,7 +8,8 @@ export const covertToDateObject: DateParser = (property, body) =>
   ['day', 'month', 'year'].reduce(
     (newDateObj: CaseDate, date: string) => {
       const propertyName = `${property}-${date}`;
-      newDateObj[date] = body[propertyName];
+      newDateObj[date] =
+        body[propertyName] && date !== 'year' ? (body[propertyName] as string).padStart(2, '0') : body[propertyName];
```

Comment:
i am assuming this has something to do with ccd and how info is stored but if so, they could be done in the toapiformat

Reply:
I've had to move this from the to-api-format, to make sure that the `session.userCase` closely matches what will be stored in the API.

This way when the user submits the "when did you get married" form, we can validate the data returned by the API making sure that it matches what's in the session to confirm that it has been saved correctly by the API.